### ISSUE TITLE
[Bug] Persisted tokenizer padding corrupts prompts in the C++ runtime

### ISSUE THREAD

**Issue by florianvoss-commit** [sima-neat/llima#139]
## Description

The C++ Hugging Face tokenizer loads `tokenizer.json` directly through tokenizers-cpp. If a checkpoint accidentally persists backend `padding` or `truncation` state, `HFTokenizer::encode()` honors that state and returns a padded/truncated sequence. LLiMa then treats padding tokens as semantic prompt tokens, runs unnecessary prefill groups, and samples logits from the padded position.

This differs from the ordinary Transformers `AutoTokenizer` call contract, which disables persisted backend padding/truncation unless requested by the caller. LLiMa already owns prompt grouping, padding to compiled group sizes, and context-length enforcement.

## Steps to Reproduce

1. Compile or obtain `simaai/Phi-4-mini-instruct-Autoround-Safetensors` at revision `ad55c6877e6eed3ef7beb3d0e4ea289e7de42d51`. Its `tokenizer.json` contains fixed right-padding and truncation at 1024 tokens.
2. Run the compiled artifact on Modalix:

   ```bash
   SIMA_LLIMA_RUN_DISABLE_QUEUE=true llima run <compiled-model-dir>
   ```

3. Enter:

   ```text
   Why is the sky blue?
   ```

4. Observe that the expected nine prompt token IDs are followed by 1,015 copies of pad token ID `199999`. The runtime executes eight n128 prefill groups (`0-128` through `896-1024`) and immediately produces repetitive garbage.

## Expected Behavior

Persisted tokenizer backend padding/truncation must not change the semantic token sequence. The prompt should contain nine tokens, execute one prefill operation for token range `0-9`, and produce coherent output.

## Environment

- Hardware platform: Modalix DevKit, aarch64
- Platform/SDK version: Neat/Modalix development environment used by LLiMa 0.4.0
- LLiMa version, commit, or package reference: reproduced with `0.4.0+develop.22d4ce59eb33`; correction validated from `develop` at `46668ae1ef4101d204ffa2e657e5cdf557bcd82d`
- Model and compilation options: `simaai/Phi-4-mini-instruct-Autoround-Safetensors`, compiled with 128-token group prefill and 2048-token context
- Additional details: tokenizers-cpp C API reads serialized backend state directly

## Logs

Before the safeguard:

```text
Formatted prompt: <|user|>Why is the sky blue?<|end|><|assistant|>
Input token ids: [200021, 13903, 382, 290, 17307, 9861, 30, 200020, 200019, 199999, ...]  # length 1024
Processing token no. 0-128
...
Processing token no. 896-1024
Assistant: ВА199: :000А : :::: ...
```

With `padding` and `truncation` cleared before constructing tokenizers-cpp:

```text
Input token ids: [200021, 13903, 382, 290, 17307, 9861, 30, 200020, 200019]
Processing token no. 0-9
Assistant: The sky appears blue due to a phenomenon called Rayleigh scattering. ...
```

TTFT decreased from approximately 0.92 seconds to 0.14 seconds.

## Additional Context

The checkpoint's quantization calibration called the fast tokenizer with `truncation=True`, `padding="max_length"`, and `max_length=1024`, then saved that same tokenizer instance. This serialized transient calibration state. A Transformers smoke test did not catch the defect because the high-level wrapper resets backend state per call.

The older published `simaai/Phi-4-mini-instruct-Autoround-a16w4` artifact contains `padding: null` and `truncation: null`, so it works without the safeguard. The minimal runtime correction is to clear these two fields before passing the JSON to tokenizers-cpp.
